### PR TITLE
Add support for the new image service ✨

### PR DESCRIPTION
**Pull request by StealWonders**
ToDo:

- [x] mutation: addImagesToReview()
- [x] mutation: deleteImageFromReview()
- [x] check if imageUrl in the Image type is still required & correct
- [x] remove image hash from DB schema
- [x] check delete error handling
- [x] add img-api url env variable
- [ ] test everything (optional) :-)

### CODE REVIEW

File: internal/graphql/resolvers/mutations.resolvers.go [hunk 1]
```diff
@@ -454,98 +516,26 @@ func (r *mutationResolver) DeleteReview(ctx context.Context, input models.Delete
...
 	if err != nil {
 		return nil, err
 	}
 
-	err = r.deleteImages(ctx, images) // if a single image fails to delete, the remaining images will be kept
-	if err != nil {
-		return nil, err
+	deletedImages := r.deleteImages(imagesUUIDs)
+	if len(deletedImages) != len(imagesUUIDs) {
+		return nil, fmt.Errorf("failed to delete all images") // dunno if we shouldn't just also return the review
```
Comment (StealWonders) [inline internal/graphql/resolvers/mutations.resolvers.go:533]: @GamerBene19 your opinion?

File: docker-compose.yml [hunk 2]
```diff
@@ -43,9 +45,8 @@ services:
       JWT_PRIVATE_KEY_PATH: /run/secrets/jwt_private_key
       JWT_ALGORITHM: RS256
       JWT_TIMEOUT_SEC: 525600
-      ASSETS_DIR: /opt/app/mensatt/assets
-      IMAGE_PROCESSOR_MAX_OUTPUT_SIZE_MB: 10
-      IMAGE_PROCESSOR_MAX_RESOLUTION: 8192
+      IMAGE_API_URL: https://localhost:3000/
+      IMAGE_API_KEY_FILE: /run/secrets/image_api_key
```
Comment (GamerBene19) [inline docker-compose.yml:49]: Wieso das Secret nicht einfach aus bspw. .env laden?

File: internal/graphql/graphql.go [hunk 3]
```diff
@@ -12,16 +12,15 @@ import (
...
-	DebugEnabled  bool
-	Database      *ent.Client
-	JWTKeyStore   *utils.JWTKeyStore
-	ImageUploader *imageuploader.ImageUploader
-	ImageBaseURL  string
+	DebugEnabled bool
+	Database     *ent.Client
+	JWTKeyStore  *utils.JWTKeyStore
+	ImageAPIURL  string
+	ImageAPIKey  string
```
Comment (GamerBene19) [inline internal/graphql/graphql.go:23]: Might just be a naming thing, but why are those two values part of the "GraphQLParams"? What effect does the image service have on GQL? Imo it should be none.

File: internal/graphql/resolvers/helper.go [hunk 4]
```diff
@@ -2,69 +2,168 @@ package resolvers
...
-	for _, image := range images {
-		if image.Image.Size > int64(r.ImageUploader.GetMaxImageSize()) {
-			return nil, fmt.Errorf("image size is too large") // let the caller handle the transaction rollback
-		}
+func (r *mutationResolver) submitImages(uuids []uuid.UUID) []uuid.UUID {
+	var submittedImages []uuid.UUID
 
-		imageBytes, err := io.ReadAll(image.Image.File)
+	for _, imageUUID := range uuids {
+		url := r.ImageAPIURL + "image/" + imageUUID.String()
```
Comment (GamerBene19) [inline internal/graphql/resolvers/helper.go:20]: Shouldn't this be submit?

File: internal/graphql/resolvers/helper.go
Comment (GamerBene19) [inline internal/graphql/resolvers/helper.go:1]: What about logging on errors?

File: internal/graphql/resolvers/helper.go [hunk 5]
```diff
@@ -2,69 +2,168 @@ package resolvers
...
-	var imageEntities []*ent.Image
-	for _, image := range images {
-		if image.Image.Size > int64(r.ImageUploader.GetMaxImageSize()) {
-			return nil, fmt.Errorf("image size is too large") // let the caller handle the transaction rollback
-		}
+func (r *mutationResolver) submitImages(uuids []uuid.UUID) []uuid.UUID {
+	var submittedImages []uuid.UUID
 
-		imageBytes, err := io.ReadAll(image.Image.File)
+	for _, imageUUID := range uuids {
```
Comment (GamerBene19) [inline internal/graphql/resolvers/helper.go:19]: Is there a way to iterate without range? I assume the ignored parameter is the idx?

File: internal/graphql/resolvers/helper.go [hunk 6]
```diff
@@ -2,69 +2,168 @@ package resolvers
...
+		url := r.ImageAPIURL + "image/" + imageUUID.String()
+		request, err := http.NewRequest("POST", url, nil)
 		if err != nil {
-			return nil, err // let the caller handle the transaction rollback
+			continue // ignore error and continue with the next image
 		}
 
-		imageUUID, imageHash, err := r.ImageUploader.ValidateAndStoreImage(imageBytes, image.Rotation)
+		request.Header.Add("Authorization", "Bearer "+r.ImageAPIKey)
+		client := &http.Client{}
```
Comment (GamerBene19) [inline internal/graphql/resolvers/helper.go:27]: Could this be moved out of the loop to avoid reassigning?

File: internal/graphql/resolvers/helper.go [hunk 7]
```diff
@@ -2,69 +2,168 @@ package resolvers
...
+
+	for _, imageUUID := range uuids {
+		url := r.ImageAPIURL + "approve/" + imageUUID.String()
+		request, err := http.NewRequest("POST", url, nil)
+		if err != nil {
+			return approvedImages, err
+		}
+
+		request.Header.Add("Authorization", "Bearer "+r.ImageAPIKey)
+		client := &http.Client{}
```
Comment (GamerBene19) [inline internal/graphql/resolvers/helper.go:66]: Could this be moved out of the loop to avoid reassigning?

File: internal/graphql/resolvers/helper.go [hunk 8]
```diff
@@ -2,69 +2,168 @@ package resolvers
...
+
+	for _, imageUUID := range uuids {
+		url := r.ImageAPIURL + "unapprove/" + imageUUID.String()
+		request, err := http.NewRequest("POST", url, nil)
+		if err != nil {
+			return unapprovedImages, err
+		}
+
+		request.Header.Add("Authorization", "Bearer "+r.ImageAPIKey)
+		client := &http.Client{}
```
Comment (GamerBene19) [inline internal/graphql/resolvers/helper.go:104]: Could this be moved out of the loop to avoid reassigning?

File: Dockerfile [hunk 9]
```diff
@@ -12,7 +12,7 @@ RUN addgroup -g $GROUP_ID mensatt && \
     adduser -D -s /bin/sh -u $USER_ID -G mensatt mensatt
 
 # This is required to set proper permissions for the assets directory when using a volume
-RUN mkdir -p /opt/app/mensatt/assets && chown -R mensatt:mensatt /opt/app/mensatt/assets
+RUN mkdir -p /opt/mensatt/assets && chown -R mensatt:mensatt /opt/mensatt/assets
```
Comment (GamerBene19) [inline Dockerfile:15]: Why is this change required?

**Review by GamerBene19 — COMMENTED** [review]
Only small changes required. Rest are only optional comments

File: docker-compose.yml (hunk 2 above)
Comment (StealWonders) [reply docker-compose.yml:49]: Weil wir das schon immer so gemacht haben

File: internal/graphql/graphql.go (hunk 3 above)
Comment (StealWonders) [reply internal/graphql/graphql.go:23]: Need to pass down config values, probably a better way to do it but this way works

File: internal/graphql/resolvers/helper.go
Comment (StealWonders) [reply internal/graphql/resolvers/helper.go:1]: Future PRs may add this

File: internal/graphql/resolvers/helper.go (hunk 5 above)
Comment (StealWonders) [reply internal/graphql/resolvers/helper.go:19]: Is this a general go question?

File: Dockerfile (hunk 9 above)
Comment (StealWonders) [reply Dockerfile:15]: It's not

**Review by stormofice — APPROVED** [review]
